.github/workflows/documentation.yml

```diff
@@ -0,0 +1,66 @@
+name: Build documentation
+
+on:
+  schedule:
+    - cron: '0 0 * * SAT' # every Saturday
+
+jobs:
+  update-docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout blog
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.USER_TOKEN }}
+
+      - name: Install OCaml
+        uses: avsm/setup-ocaml@v1
+        with:
+          ocaml-version: 4.09.1
+
+      - name: Pin OASIS
+        run: opam pin add oasis https://github.com/BinaryAnalysisPlatform/oasis.git
+
+      - name: Checkout bap
+        uses: actions/checkout@v2
+        with:
+          repository: BinaryAnalysisPlatform/bap
+          path: bap.master
+
+      - name: Pin BAP
+        run: opam pin add bap bap.master --no-action
+
+      - name: Install system dependencies
+        run: opam depext -u bap
+
+      - name: Install opam dependencies
+        run: |
+            opam install bap --deps-only
+            opam install omake
+
+      - name: Install Bap
+        run: |
+          cd bap.master
+          opam exec -- ./configure-omake
+          opam exec -- make
+          opam exec -- make reinstall
+          cd ../
+
+      - name: Build Documentation
+        run: |
+          sudo apt-get install emacs-nox man2html --yes
+          opam install odig --yes
```

Comment:
`--yes` is not needed in for opam as `OPAMYES` is set to true by the setup-ocaml action.